server/src/main/java/game/snake/Snake.java

```diff
@@ -102,6 +107,12 @@ public void tick() {
         }
 
         updateWidth();
+        var currentWorldChunk = world.chunks.findChunk(headPosition);
+        boolean worldChunkChanged = lastWorldChunk != currentWorldChunk;
```

Comment:
This variable is not required, it can be inlined into the if statement (the only usage). Also be aware that `!=` compares references (pointers) which actually works in this case but generally check for object equality with `Objects.equals(a,b)`.